src/herder/Upgrades.cpp

```diff
@@ -314,6 +453,12 @@ Upgrades::removeUpgrades(std::vector<UpgradeType>::const_iterator beginUpdates,
         resetParamIfSet(res.mBaseReserve);
         resetParamIfSet(res.mFlags);
 
+        if (res.getConfigUpgradeSet())
```

Comment:
Missed this one earlier: `mParams.mConfigUpgradeSet` is basically a cache, so every time we close a ledger we have to make sure everything is still valid, including that we can rebuild from just the `ConfigUpgradeSetKey` (in case the ledger entry that contains the pre-image gets deleted/modified).

Given how the code is now factored and to avoid more of those edge cases, maybe we should get rid of that cache and only store the `ConfigUpgradeSetKey` in `mUpgrades` and just rebuild `ConfigUpgradeSetFrame` every time we need it?

Reply:
Removed ConfigUpgradeSet from mUpgrades.